### PR TITLE
Port to meson build system

### PR DESCRIPTION
**Pull request by tintou**
This is a meson port which might help to maintain PackageKit
(follow-up of #307)

### CODE REVIEW

**Comment by hughsie** [comment]
Ninja dist seems to fail now:

    ERROR: ['gtkdoc-mkhtml', '--path=/tmp/tmpoehjxhle/PackageKit-1.1.13/docs/api:/tmp/tmpx1bl9br8/docs/api', 'PackageKit', '../PackageKit-docs.sgml'] failed with status 6
    warning: failed to load external entity "../dbus/org.freedesktop.PackageKit.ref.xml"
    ../PackageKit-docs.sgml:57: element include: XInclude error : could not load ../dbus/org.freedesktop.PackageKit.ref.xml, and no fallback was found
    warning: failed to load external entity "../dbus/org.freedesktop.PackageKit.Transaction.ref.xml"
    ../PackageKit-docs.sgml:58: element include: XInclude error : could not load ../dbus/org.freedesktop.PackageKit.Transaction.ref.xml, and no fallback was found

Also, aren't meson options supposed to be using `_` rather than `-` these days?

**Comment by hughsie** [comment]
> GNOME Builder uses underscores

I was about to say that fwupd uses underscores, but now I see it uses both :/ -- I think there was some pressure for gnome-software to move from - to _, so I think that's probably what we should do here. Sorry!

File: contrib/PackageKit.spec.in [hunk 1]
```diff
@@ -135,29 +135,19 @@ using PackageKit.
...
 %else
-        --disable-bash-completion \
+        --Dbash-completion=false \
 %endif
-        --disable-local \
-        --disable-silent-rules
-
-make %{?_smp_mflags} V=1
+        -Dlocal-checkout=false
+meson_build
```
Comment (Conan-Kudo) [inline contrib/PackageKit.spec.in]: `meson_build` -> `%meson_build`

Comment (tintou) [reply contrib/PackageKit.spec.in]: Fixed

**Comment by hughsie** [comment]
I think get the tests fixed on CI, and then we can merge this. Thanks.

File: tests/ci/Dockerfile-fedora [hunk 2]
```diff
@@ -3,6 +3,7 @@ FROM fedora:29
 RUN dnf -y update
 RUN dnf -y install dnf-plugins-core libdnf-devel redhat-rpm-config autoconf-archive gcc make
```
Comment (Conan-Kudo) [inline tests/ci/Dockerfile-fedora]: Drop `autoconf-archive` and `make` and put `meson` and `ninja-build` here.

File: tests/ci/Dockerfile-fedora [hunk 3]
```diff
@@ -3,6 +3,7 @@ FROM fedora:29
...
 RUN dnf -y builddep PackageKit
+RUN dnf -y install meson
```
Comment (Conan-Kudo) [inline tests/ci/Dockerfile-fedora]: This will not be needed once the above is fixed.

File: tests/ci/Dockerfile-fedora [hunk 4]
```diff
@@ -1,9 +1,8 @@
 FROM fedora:29
 
 RUN dnf -y update
-RUN dnf -y install dnf-plugins-core libdnf-devel redhat-rpm-config autoconf-archive gcc make
+RUN dnf -y install dnf-plugins-core libdnf-devel redhat-rpm-config meson gcc inja-build
```
Comment (belka-ew) [inline tests/ci/Dockerfile-fedora]: should be **n**inja-build

**Comment by t-chaik** [comment]
@tintou: Are you planning on updating this / addressing the last comments? Two GNOME goals needs it: the [gettext port](https://wiki.gnome.org/Initiatives/GnomeGoals/GettextMigration) and the [Meson port](https://wiki.gnome.org/Initiatives/GnomeGoals/MesonPorting). Would be nice to have that included in next PackageKit release. I could help if you don't have time to spend on this patch set any more.

File: tests/ci/Dockerfile-fedora (hunk 4 above)
Comment (tintou) [reply tests/ci/Dockerfile-fedora]: thanks

**Comment by tintou** [comment]
@t-chaik Yeah, feel free to continue the work

**Comment by jjardon** [comment]
@t-chaik hey! did you have, by any chance, time to update this PR?

**Comment by t-chaik** [comment]
Haven't been able to find some time yet but that's still on my list. I'll try to have a look next week. Would that work for your release or would it be too late?

**Comment by t-chaik** [comment]
@hughsie: Do you mind giving me write access to the `wip/tintou/meson` branch? I can create a new PR if you prefer, up to you.

**Comment by Conan-Kudo** [comment]
@t-chaik Could you create a new PR initially? That would unblock showing us the code.

**Comment by hughsie** [comment]
@t-chaik i think a new PR is best.

**Comment by t-chaik** [comment]
Here: https://github.com/hughsie/PackageKit/pull/332. I've pushed some fixes already but haven't look at the test failures yet. Will remove the WIP label once I believe it to be ready.

**Comment by hughsie** [comment]
I've fixed up a lot of the issues, so lets continue in https://github.com/hughsie/PackageKit/pull/365 -- all comments there very welcome, thanks!

**Pull request closed without merging**